shell/platform/android/io/flutter/view/AccessibilityBridge.java

```diff
@@ -558,7 +558,7 @@ public AccessibilityNodeInfo createAccessibilityNodeInfo(int virtualViewId) {
     if (semanticsNode.platformViewId != -1) {
       View embeddedView =
           platformViewsAccessibilityDelegate.getPlatformViewById(semanticsNode.platformViewId);
-      boolean childUsesVirtualDisplay = !(embeddedView.getContext() instanceof FlutterActivity);
+      boolean childUsesVirtualDisplay = embeddedView.getContext() instanceof PresentationContext;
```

Comment:
This isn't very robust, nothing stops platform views from providing their own contexts. Can we track the platform view type within the embedding code so that `childUsesVirtualDisplay` can't be changed by plugin code?

Reply:
Under what situation someone just ignores the context passed to the factory when creating the view?  `PresentationContext` has some logic like `isCalledFromAlertDialog`, does it mean that we support not using that functionality?

Reply:
One doesn't have to ignore the context passed in, Android provides [ContextWrapper](https://developer.android.com/reference/android/content/ContextWrapper) which lets you change just the thing you need in the context. If that is done the instanceOf check will not check for what you want.

Reply:
so given that we will deprecate this code in the near future, what about we use `ContextWrapper#getBaseContext()` to get the original context?

Reply:
What's the deprecation timeline?
Is it hard to track a bit for whether this is a HC or VD platform view?

Reply:
Deprecation timeline is Q1 if the g3 customers are migrated by then. (they almost are)

I looked at the code, it's not hard to expose it. Most of the work is ensuring that the test is actually testing something. Do you want to go that route?

Reply:
If it's not much work I rather take the robust approach. Especially if we can't guarantee a deprecation date.

Reply:
sure. My question is the issue we are trying to address is accessibility in virtual displays, which is broken.  I can't get webview to do basic things like exit the webview and announce Flutter UI. What is the value added by making this more robust, if it won't work anyways?

Reply:
If it's completely broken and we don't want to fix it we should remove the a11y completely.
Is this really the case though? AFAIK `customer: dream` have performed pre-launch a11y tests which have passed sufficiently and are currently relying on this. Are you saying we broke them completely at some point?

Reply:
BTW isn't the robustness of this check is important for both the VD and HC use cases?

Reply:
On Android 10, b/171449547 is broken if the view is in a virtual display. 

For HC, this check should always work since `PresentationContext` isn't a public API.  Once VD is deprecated, this check will not exist as platform views using VD will use HC automatically.     Deprecating VD doesn't break public APIs, we just would just need to change the implementation of `AndroidView` in the framework.

Reply:
Good point about PresentationContext not being public (though technically you just turned it public in this PR, is `@hide` for non-Android-SDK code enforces anything or does it just prune Javadocs?)
I see your point that this is *likely* not going to break folks anytime soon, however assuming a robust fix (keeping track of the kind of view) requires no more than a couple hours of work my preference would be to keep on the safe side.
My concern would be that we will end up staying with VDs for longer than we think right now and some non standard use-case (of a replaced context) will turn into a debugging nightmare.

Reply:
The patch ended up being smaller. PTAL.

`@hide` would hide it from the public documentation. I wish it had the same meaning it has for the Android-SDK code :)

Reply:
Shorter than this thread 😄 

Thanks!